### PR TITLE
[Ready]Fan HUD

### PR DESCRIPTION
**Pull request by RaveRadbury**
## About The Pull Request
Building off of #47254, this PR adds a HUD for Clown fans and Mime fans (along with clowns and mimes) that allow them to identify fan-pin wearers on-sight via a HUD. It also keeps pins from showing through on exosuits, as there is now a better visual indicator.

image 
image

## Why It's Good For The Game
The purpose of the fan quirks is to instill a sense of tribalism similar to sports. In-group out-group behaviors are largely based on visual recognition. The pins on-uniform haven't been as easy to see as hoped, resulting in players taking off their pins to show directly to their preferred performer. This has resulted in crew not being very aware of their opposing fanbase. With the Fan HUD, all involved should have a much easier time identifying who is wearing a mime pin and who is wearing a clown pin.

## Changelog
:cl:
add: Quirky Fans can now identify clown/mime pins on-sight
tweak: Fan pins no longer show over exosuits
/:cl:

Big thanks to 
@Whoneedspacee for overall coding help and debugging
@Krysonism for the nice HUD icons

### CODE REVIEW

**Comment by itseasytosee** [comment]
Huge buff to ghetto revs

**Comment by optimumtact** [comment]
`The purpose of the fan quirks is to instill a sense of tribalism similar to sports. `
https://www.youtube.com/watch?v=w35JarUKT6w

**Comment by wesoda25** [comment]
I don’t like this because if you want roleplay between fellow fans it should be natural and not “oh you got the hud thing whats up”

**Comment by Whoneedspacee** [comment]
I feel like it’d be a neat side minigame and only barely round affecting which is what good quirks aim to deliver imo.

**Review by Twaticus — APPROVED** [review]

File: code/game/data_huds.dm [hunk 1]
```diff
@@ -210,6 +213,23 @@
...
+/mob/living/carbon/human/proc/fan_hud_set_fandom()
+	var/image/holder = hud_list[FAN_HUD]
+	var/icon/I = icon(icon, icon_state, dir)
+	holder.pixel_y = I.Height() - world.icon_size
+	holder.icon_state = "hudfan_no"
+	var/obj/item/clothing/under/U = get_item_by_slot(ITEM_SLOT_ICLOTHING)
+	if(U && istype(U.attached_accessory, /obj/item/clothing/accessory/fan_mime_pin))
+		holder.icon_state = "fan_mime_pin"
+	else if (U && istype(U.attached_accessory, /obj/item/clothing/accessory/fan_clown_pin))
+		holder.icon_state = "fan_clown_pin"
```
Comment (Rohesie) [inline code/game/data_huds.dm:231]:
```suggestion
	if(U)
		if(istype(U.attached_accessory, /obj/item/clothing/accessory/fan_mime_pin))
			holder.icon_state = "fan_mime_pin"
		else if (istype(U.attached_accessory, /obj/item/clothing/accessory/fan_clown_pin))
			holder.icon_state = "fan_clown_pin"
```

**Comment by Rohesie** [comment]
This is pretty minor.
The hud only appears when you are wearing the pin, so it's just made more visible.
I'll merge it once the change is done unless if someone vetoes it in the meantime.

**Comment by RaveRadbury** [comment]
> This is pretty minor.
> The hud only appears when you are wearing the pin, so it's just made more visible.
> I'll merge it once the change is done unless if someone vetoes it in the meantime.

I just tried it and it seemed to work when I placed in on someone else. 
image

The block in question is concerned with setting the icon state, rather than allowing the HUD. 

Am I misunderstanding something?

**Comment by Rohesie** [comment]
I mean that the hud is just a way to make the pin more visible, it goes away if the pin is removed, and it returns once it's back. Which is fine, not very controversial.

**Comment by RaveRadbury** [comment]
> I mean that the hud is just a way to make the pin more visible, it goes away if the pin is removed, and it returns once it's back. Which is fine, not very controversial.

This is intended. The idea is that fans are recognizing merch from their fandom and a rival fandom when it is worn on anyone's jumpsuit. 

Only fans and performers can see the HUD, but the icon state should be getting set for anyone wearing the pins. Wearing a pin should not grant you access to the HUD, only cause you to appear on it.

**Comment by RaveRadbury** [comment]
It's not accepting new commits what is happening to my world.

**Pull request closed without merging**